tests/unit_tests/test_tallies.py

```diff
@@ -39,3 +39,6 @@ def test_xml_roundtrip(run_in_tmpdir):
     assert new_tally.triggers[0].trigger_type == tally.triggers[0].trigger_type
     assert new_tally.triggers[0].threshold == tally.triggers[0].threshold
     assert new_tally.triggers[0].scores == tally.triggers[0].scores
+    # delete tally and check that new_tallies now has length zero
+    new_tally.openmc_remove_tally_from_tallies(tally.id)
```

Comment:
This test performs checks on the pure Python API as opposed to the `openmc.lib` module, which is the Python interface to the CAPI. This check looks good to me, but we'll want to add something similar in `tests/unit_tests/test_lib.py`.

Reply:
Would it make sense to just cut these lines from `test_tallies.py` and do it instead in `test_lib.py` with a similar setup?